scripts/Util.py

```diff
@@ -2314,19 +2314,29 @@ def setProcessController(self, proxy, current):
         import Ice
 
         comm.getProperties().setProperty("Adapter.AdapterId", str(uuid.uuid4()))
-        self.adapter = comm.createObjectAdapterWithEndpoints("Adapter", endpoints)
-        self.adapter.add(
-            ProcessControllerRegistryI(self),
-            Ice.stringToIdentity("Util/ProcessControllerRegistry"),
-        )
-        self.adapter.activate()
+        if endpoints:
+            print("starting process controller registry on {0}".format(endpoints))
+            self.adapter = comm.createObjectAdapterWithEndpoints("Adapter", endpoints)
+            self.adapter.add(
+                ProcessControllerRegistryI(self),
+                Ice.Identity(category="Util", name="ProcessControllerRegistry"),
+            )
+            self.adapter.activate()
 
     def __str__(self):
         return "remote controller"
 
     def getHost(self, current):
         return self.getController(current).getHost(current.config.protocol, current.config.ipv6)
 
+    def getControllerIdentity(self, current):
```

Comment:
Not from this PR: we should definitely never use the parameter name `current` for anything other than Ice.Current.